### PR TITLE
feature/35_SHOW_LISTING

### PR DESCRIPTION
**Pull request by Symphonie234**
- Added functionality for liking and setting favorite of the listing
- Added a blade file for showing the content of the Listing

### CODE REVIEW

File: app/Models/Listing.php [hunk 1]
```diff
@@ -168,12 +168,22 @@ public function unlike(User $user)
...
     {
         $likeCount = $this->likes->count();
+        $currentUser = auth()->user();
 
         if ($likeCount === 0) {
             return 'No one liked this post yet';
-        } elseif ($likeCount <= 2) {
-            $likedBy = $this->likes->pluck('user.name')->implode(', ');
-            return "$likedBy liked this post";
+        } elseif ($likeCount === 1 && $this->likes->contains('user_id', $currentUser->id)) {
```
Comment (ChronoDevs) [inline app/Models/Listing.php]: make an enum for $likeCount values

File: app/Http/Controllers/FavoriteController.php [hunk 2]
```diff
@@ -79,8 +79,9 @@ public function addToFavorites(Listing $listing)
 
         if (!$user->favorites()->where('listing_id', $listing->id)->exists()) {
             $user->favorites()->create(['listing_id' => $listing->id]);
+            return back()->with('success', 'Listing added to favorites!');
         }
 
-        return redirect()->back();
+        return back()->with('error', 'Listing is already in favorites!');
```
Comment (ChronoDevs) [inline app/Http/Controllers/FavoriteController.php]: use custom messages from lang

**Comment by Symphonie234** [comment]
### Changes that have been addressed in the PR No. 1:

- Created a new **Enums** file named **`LikeCount`**.
- Added a custom language for the **`FavoriteController`**.

### Additional Changes:

- Adjusted the **`FavoriteController`** and **`ListingController`**.
- Modified the **`Listing`** Model.
- Utilized **Spatie/Enum**.
- Included new aliases and providers in the **`config/auth.php`** file.
- Added a new JavaScript file named **`listings_show.js`** and incorporated new styles in the **`listings_show.css`**.
- Updated blade files to accommodate the changes.

File: app/Models/Listing.php (hunk 1 above)
Comment (ChronoDevs) [reply app/Models/Listing.php]: also message use lang

**Comment by Symphonie234** [comment]
### Changes that have been addressed in the PR No. 2:

- Created another **lang/en** file named **`listing.php`** to include specific messages for likes.
- Adjusted the code in the **`Listing`** Model accordingly.

File: app/Models/Listing.php (hunk 1 above)
Comment (ChronoDevs) [reply app/Models/Listing.php]: sorry the $likeCount condition should be check from enum
if ($likeCount === LikeCount::zero) { 

use your enum from 
class LikeCount extends Enum

Comment (ChronoDevs) [reply app/Models/Listing.php]: after fix. Please merge. Thank you

**Comment by Symphonie234** [comment]
### Changes that have been addressed in the PR No. 3:

- Adjusted the code of the **`getLikesAttribute`** function in the **`Listing`** Model.

**Pull request merged**